### PR TITLE
feat: implement client credential flow in auth2

### PR DESCRIPTION
**Pull request by danielgent**
Lots of copy-pasting from the token service :smile: 

image

### CODE REVIEW

File: src/token-grant-types/authorizationCodeGrant.ts [hunk 1]
```diff
@@ -52,3 +54,71 @@ export async function authorizeCodeGrant(
...
+
+export async function clientCredentialsGrant(
+  env: Env,
+  params: ClientCredentialGrantTypeParams,
+  // on token service this is | Null instead...
+): Promise<TokenResponse | CodeResponse> {
+  const client = await getClient(env, params.client_id);
+
+  // Validate the secret
+  // const secretHash = await hash(params.client_secret);
```
Comment (danielgent) [inline src/token-grant-types/authorizationCodeGrant.ts:66]: Are we planning to store the secrets hashed?  

The token service stores a list of secrets

File: src/token-grant-types/authorizationCodeGrant.ts [hunk 2]
```diff
@@ -52,3 +54,71 @@ export async function authorizeCodeGrant(
...
+  // Validate the scopes
+  // we don't have scopes on auth2 do we?
+  // await checkScopes({
+  //   env,
+  //   client,
+  //   requestedScopes: params.scope.split(" "),
+  // });
+
+  // same... we don't have this right?
+  // const permissions = await getPermissions({ env, client });
```
Comment (danielgent) [inline src/token-grant-types/authorizationCodeGrant.ts:86]: There are no scopes or permissions on Auth2 so should I delete this code for now? :thinking:

File: src/token-grant-types/authorizationCodeGrant.ts (hunk 1 above)
Comment (markusahlstrand) [reply src/token-grant-types/authorizationCodeGrant.ts:65]: Good question. Think the secrets are readable in the auth0 dashboard so maybe we should keep them plain?

File: src/token-grant-types/authorizationCodeGrant.ts (hunk 2 above)
Comment (markusahlstrand) [reply src/token-grant-types/authorizationCodeGrant.ts:85]: Yepp, think so. Unless we need it to make the auth0 management api happy.

**Review by markusahlstrand — APPROVED** [review]
LGTM

**Pull request merged**